Application/src/view/visuals/component/Message.java

```diff
@@ -90,76 +84,81 @@ public boolean isKeepAnimating() {
-                        keepAnimating = false;
-                    }
-                    break;
-
-                //default case
-                default :
-                    selfCallCounter =1;
-            }
-        }
+        int move = class_size/6;
 
+        // If we are animating
+        if (animating) {
 
-           // If it's being viewed in the past and not currently animating.
-           if (staticIndicator) {
-
-               // Remove all trails
-               trails.clear();
-
-               // Which direction is it pointing at determines original location.
-               animationBounds = 0;
-
-               // Set it's animation state to true.
-               keepAnimating = true;
+            //Checks if we are supposed to keep animating, set animationBounds according to how diagramClasses are scaled.
 
-               //Check if the message is a self-call and reset the offset.
-               if(fromNode == toNode) {
-                   offset -= 24;
-               }
 
-               //Resets counter for self-calls.
-               selfCallCounter = 1;
+            // Sending a message.
+            if(node2.getX() > node1.getX()) {
+
+                if ((animationBounds += move) > this.node2.getX() - this.node1.getX()) {
+                    animating = false;
+                }
+
+                // Sending a return message.
+            } else if(node1.getX() > node2.getX()){
+
+                if((animationBounds -= move) < (this.node2.getX())  - this.node1.getX()) {
+                    animating = false;
+                }
+
+                // Self calls
+            } else /*if (node1.getX() == node2.getX())*/ {
+
+                switch (selfCallCounter) {
+                    case 1:
+                        // Move the message to the right of the class
+                        if ((animationBounds += move) > this.class_size) {
+                            // Original state of the dragon's flying direction
+                            directionSwitched = !directionSwitched;
+                            selfCallCounter = 2;
+                        }
+                        break;
+
+                    case 2:
+                        // Multiple increments to lower the message vertically
+                        offset += 24;
+                        selfCallCounter = 3;
+                        break;
+
+                    case 3:
+                        // move back to starting point
+                        if ((animationBounds -= move) < 1) {
+                            animating = false;
+                        }
+                        break;
+
+                    //default case
+                    default :
+                        selfCallCounter = 1;
+                }
+            }
+        }
 
-               // Resets static indicator.
-               staticIndicator = false;
+        // If it's being viewed in the past and not currently animating.
+        if (staticIndicator) {
```

Comment:
I still think this variable has a vague name; also static means something else in Java

Reply:
+1

Reply:
This was added 3 sprints ago.

Reply:
It was not an issue then?

Reply:
if we're improving the code, might as well change it now is all